datadog/resource_datadog_authn_mapping.go

```diff
@@ -0,0 +1,203 @@
+package datadog
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/DataDog/datadog-api-client-go/api/v2/datadog"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/terraform-providers/terraform-provider-datadog/datadog/internal/utils"
+)
+
+func resourceDatadogAuthnMapping() *schema.Resource {
+	return &schema.Resource{
+		Description:   "Provides a Datadog AuthN Mappings resource. This can be used to create and manage Datadog SAML Mappings.",
+		CreateContext: resourceDatadogAuthnMappingCreate,
+		ReadContext:   resourceDatadogAuthnMappingRead,
+		UpdateContext: resourceDatadogAuthnMappingUpdate,
+		DeleteContext: resourceDatadogAuthnMappingDelete,
+
+		Schema: map[string]*schema.Schema{
+			"key": {
+				Description: "Identity provider key.",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"value": {
+				Description: "Identity provider value.",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"role": {
+				Description: "The role to assign for key:value mapping.",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+		},
+	}
+}
+
+func resourceDatadogAuthnMappingCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*ProviderConfiguration).DatadogClientV2
+	auth := meta.(*ProviderConfiguration).AuthV2
+	authNMapReq := buildAuthNMappingCreateRequest(d)
+	createResp, httpResponse, err := client.AuthNMappingsApi.CreateAuthNMapping(auth, authNMapReq)
+	if err != nil {
+		return utils.TranslateClientErrorDiag(err, httpResponse, "error creating authn mapping")
+	}
+	if err := utils.CheckForUnparsed(createResp); err != nil {
+		return diag.FromErr(err)
+	}
+
+	var getAuthNMappingResponse datadog.AuthNMappingResponse
+	var httpResponseGet *http.Response
+
+	err = resource.RetryContext(ctx, d.Timeout(schema.TimeoutCreate), func() *resource.RetryError {
```

Comment:
Curious why Create gets wrapped with a retry but other operations don't. Is that some convention you copied from elsewhere in this repo, or did you run into failures/timeout and need to add retries?